woocommerce/woocommerce
# Coming soon: Do not render header and footer if the theme does not support FSE

…current theme is not a fse

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #50915  .

This PR fixes double rendering of the header and footer sections when the current theme does not support FSE and the `store pages only` option is enabled for the coming soon mode.

### Cause

When the current theme does not support FSE and the `store pages only` option is enabled, we defer the rendering the header and footer to the theme by calling [get_header](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php#L70) and [get_footer](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php#L85). However, we also render the `coming-soon` content in the [template-canvas.php](https://github.com/WordPress/WordPress/blob/master/wp-includes/template-canvas.php), which includes both the header and footer.

Note: Calling [get_query_tempalte( 'coming-soon' )](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php#L67) returns template-canvas.php. This is expected.

![Screenshot 2024-09-23 at 2 01 43 PM](https://github.com/user-attachments/assets/5bf67de3-1b88-469d-9a43-15c99f679852)


We can fix it by rendering the coming soon template content without the header and footer part when the theme is not FSE and the store pages only option is enabled.





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Install and activate Code Snippets plugin.
3. Go to `Snippets -> Add New` and create a new snippet with the following code. Make sure to enable the snippet.

```
function test() { echo 'test-head'; }
add_action('wp_head','test');
```
4. Install and activate Twenty Twenty-One theme (non-FSE theme).
5. Go to `WooCommerce -> Settings -> Site Visibility`
6. Enable `Coming soon` with `Apply to store pages only` option.
7. Open a new incognito window and navigate to https://YOUR-SITE/shop
8. You should see only one `test-head` text.
 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Fix double rendering of the header and footer when the coming soon mode is enabled with the store pages only option with a non-FSE theme.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
